Source/Dafny/AST/DafnyAst.cs

```diff
@@ -5759,8 +5787,48 @@ public class SubsetTypeDecl : TypeSynonymDecl, RedirectingTypeDecl {
     public enum WKind { CompiledZero, Compiled, Ghost, OptOut, Special }
     public readonly SubsetTypeDecl.WKind WitnessKind;
     public readonly Expression/*?*/ Witness;  // non-null iff WitnessKind is Compiled or Ghost
-    public bool ConstraintIsCompilable; // Will be filled in by the Resolver
-    public bool CheckedIfConstraintIsCompilable = false; // Set to true lazily by the Resolver when the Resolver fills in "ConstraintIsCompilable".
+    bool ConstraintIsCompilable; // Will be filled in by the Resolver
+    bool ConstraintIsCompilableComputed = false; // Set to true lazily by the Resolver when the Resolver fills in "ConstraintIsCompilable".
+    public string ConstraintIsCompilableReasonIfNot;
+    public IToken ConstraintIsCompilableLocationIfNot;
```

Comment:
Yes it can be accessed more than once. For every comprehension, it will verify that the subset constraint can be satisfied after the range, and it generates an assertion error accordingly.

Reply:
I haven't merged yet the error capturing datatypes, but error capturing datatypes. Let me explore this idea after today when I will have merged master into it.